text/0419-cdk-launchpads.md

```diff
@@ -0,0 +1,539 @@
+          "Ref": "PermissionBoundaryPolicy"
+        }
+      }
+    }
+  }
+}
+```
+
+And also with CDK, using the same CDK libraries that are used for applications:
+
+```ts
+import * as iam from 'aws-cdk-lib/aws-iam';
+
+const managedPolicy = new iam.ManagedPolicy(launchpad, 'Boundary', {
+  statements: [
+    new iam.PolicyStatement({
+      effect: iam.Effect.DENY,
+      actions: ['iam:'],
+      resources: [''],
+    }),
+  ],
+});
+// surface this resource to application developers using this Launchpad
+// (seee below)
+launchpad.export(managedPolicy);
+```
+
+#### IAM Permission Boundaries
+
+Launchpads allow setting a given Managed Policy as the Permissions Boundary to be used by every application deployed into a given environment.
+
+You can do it through JSON/YAML:
+
+```json
+{
+  "launchpad": {
+    "iam": {
+      "permissionBoundary": {
+        "Ref": "PermissionBoundaryPolicy"
+      }
+    }
+  }
+}
+```
+
+Or in CDK code:
+
+```ts
+launchpad.setPermissionBoundaryPolicy(managedPolicy);
+```
+
+### Importing the Launchpad in CDK code
+
+Once a Launchpad has been deployed by the administrators,
+application developers can import it using the
+[`cdk-import` tool](https://github.com/cdklabs/cdk-import)
+(whose functionality will be incorporated into the main CDK CLI in the future).
+
+Executing the following command:
+
+```shell
+$ cdk-import launchpad -l typescript
+```
+
+Will generate code in your project that you then use in your CDK app definition:
+
+```ts
+import * as cdk from 'aws-cdk-lib';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import Launchpad from '../build';
+
+const app = new cdk.App({
+    launchpad: Launchpad, // adds the appropriate Permission Boundary automatically
+});
+const stack = new cdk.Stack(app, 'MyStack');
+const fn = new lambda.Function(stack, 'MyLambda', {
+  code: new lambda.Code.fromAsset('lambda-code'),
+  handler: 'index.handler',
+  runtime: lambda.Runtime.PYTHON_3_9,
+  // refer to a resource defined in the Launchpad
+  role: Launchpad.Resources.functionRole,
+});
+```
+
+Additionally, importing the Launchpad in that way will produce a warning during `cdk deploy`
+if the version of the Launchpad in your environment is older than the current latest version of the Launchpad,
+and will ask you to update.
+
+### Integrating Launchpads with AWS Organizations
+
+Launchpads will integrate with AWS Organizations.
+When creating a new account using the AWS Console for Organizations,
+you'll be able to specify which Launchpad that account should use.
+This means the account will be provisioned ready for CDK deployments from the start -
+there will be no need for any manual actions, like running `cdk bootstrap`.
+
+## Public FAQ
+
+### What are we launching today?
+
+We are launching CDK Launchpads,
+a product that allows platform administrators in enterprise environments to easily configure AWS environments for deploying CDK applications.
+
+### Why should I use this feature?
```

Comment:
Can platform administrators use Launchpad for applications deployed using other provisioning engines? Why does the RFC mention CDK applications only? Does it mean the platform administrators will need to invest a separate effort for other provisioning engines?

Reply:
The part that is CDK-specific is the bootstrap resources that are used are CDK-specific.

Other provisioning engines could be supported by customizing the Launchpad shared resources, I guess. I'll try to add it to the doc.

Reply:
@skinny85 thanks! Thinking about it, what is CDK-specific about bootstrap resources? The way I think about is that these resources support a certain design approach, and CDK makes use of them to implement it. Other tools might use these resources similarly, or extend them through customization (as you mentioned).

Reply:
I mean, it's really hard to judge in advance whether other tools could use these resources in the same way as CDK does, without doing some sort of dive-deep into a specific tool 🙂.

Taking a step back, the bootstrap resources CDK needs are:
1. An S3 Bucket (along with a BucketPolicy) and ECR repository (for Asset support).
2. Several IAM Roles (for Asset publishing, for CDK context lookups, and also for CloudFormation deployments, including cross-account deployments).
3. Optional KMS Keys for encrypting the Bucket and the repo, and Aliases to make these Keys easier to manage in the Console.
4. An SSM parameters that stores the current version of the bootstrap Stack.
5. All of the above resources (besides the KMS Keys) need to have a set physical name according to a specific convention, otherwise CDK will not be able to use them.
6. We need a few CloudFormation Outputs (again, with specific physical names), used the CDK CLI.

That's a pretty specific list! Of course, I'm not saying none of these can be used by other tools - but looking at how specific the CDK requirements are, it's very likely other tools will have similarly detailed requirements of their own, and re-using the same resources for both will be tricky.

Reply:
It makes sense and seems like a 2-way door :)